### ISSUE TITLE
Unexpected errors when run SQL lint on Athena queries

### ISSUE THREAD

**Issue by hongbo-miao** [sqlfluff/sqlfluff#5155]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

I got unexpected errors when run SQL lint on Athena queries.

### Expected Behaviour

No errors.

### Observed Behaviour

Has errors.

### How to reproduce

I have a file **query.sql**:

```sql
select
    timestamp,
    current,
    voltage,
    temperature
from awsdatacatalog.hm_iot_db.motor limit 100;

show functions;

describe awsdatacatalog.hm_iot_db.motor;

show create table awsdatacatalog.hm_iot_db.motor;
```

`sqlfluff lint --dialect=athena query.sql` gives me results

```shell
L:   1 | P:   1 | LT09 | Select targets should be on a new line unless there is
                       | only one select target.
                       | [layout.select_targets]
L:   3 | P:   5 |  PRS | Line 3, Position 5: Found unparsable section:
                       | 'current,\n    voltage,\n    temperature'
L:   8 | P:   1 |  PRS | Line 8, Position 1: Found unparsable section: 'show
                       | functions'
L:  10 | P:   1 |  PRS | Line 10, Position 1: Found unparsable section: 'describe
                       | awsdatacatalog.hm_iot_db.motor'
WARNING: Parsing errors found and dialect is set to 'athena'. Have you configured your dialect correctly?
```

### Dialect

athena

### Version

2.3.1

### Configuration

None

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by peterbud** [comment]
The above statement is not a valid Athena dialect, as it contains two reserved keywords which cannot be used without quotes. See reserved keywords: https://docs.aws.amazon.com/athena/latest/ug/reserved-words.html
